ChartIQ/finsemble-seed
# 19159 [FOR REVIEW ONLY] Download management POC

FOR REVIEW ONLY: #19159 [CARD]

[Kanban link](https://chartiq.kanbanize.com/ctrl_board/48/cards/19159/details/)

**Description of change**
1. Added 2 sample components to illustrate 2 download style type
- Chrome Style: Each separate components handle its download information
- Firefox: A download manager to manage all download information from other components.

**Description of testing**
i.e. Please use the following branch of Finsemble and FEA
https://github.com/ChartIQ/finsemble/tree/poc/citi_download_feature_request
https://github.com/ChartIQ/finsemble-electron-adapter/tree/poc/citi_download_feature_request

Test for Chrome style download
1. Spawn 1 "chromeDownload" component
2. Press "Download Sample xls file" button to initiate the download
3. Observe the change in file list 
4. Press "Download Sample xlsx file" button to initiate another download
5. Folder selection prompt should prompt to select a folder
6. Observe the change in file list
7. Change the download config in both components.json and manifest-local.json. Detailed available options are illustrated in the comments of both files.
8. Run through step 1-6 to observe the change of behaviour after changing the config. 
 
Test for Firefox style download
1. Spawn 1 "firefoxDownloadManager" component and 2 "chromeDownload" components.
2. Press "Download Sample xls file" button to initiate the download in both "chromeDownload" components.
3. Observe the change in file list in "firefoxDownloadManager" 
4. Press "Download Sample xlsx file" button to initiate another download
5. Folder selection prompt should prompt to select a folder
6. Observe the change in file list in "firefoxDownloadManager"
